arangod/RocksDBEngine/RocksDBHotBackup.cpp

```diff
@@ -106,75 +109,60 @@ std::shared_ptr<RocksDBHotBackup> RocksDBHotBackup::operationFactory(
 //
 // @brief Setup the base object, default is "bad parameters"
 //
-RocksDBHotBackup::RocksDBHotBackup(const VPackSlice body)
+RocksDBHotBackup::RocksDBHotBackup(VPackSlice body)
   : _body(body), _valid(false), _success(false), _respCode(rest::ResponseCode::BAD),
-    _respError(TRI_ERROR_HTTP_BAD_PARAMETER), _timeoutSeconds(10)
-{
-  return;
-}
+    _respError(TRI_ERROR_HTTP_BAD_PARAMETER), _timeoutSeconds(10) {}
 
-//
-// @brief
-//
-RocksDBHotBackup::~RocksDBHotBackup() {};
-
-std::string RocksDBHotBackup::buildDirectoryPath(const std::string & timestamp, const std::string & userString) {
-  std::string ret_string, suffix;
```

Comment:
In general I agree that we should look for RVO opportunities, but not here.
That code path is probably called a handful of times during the lifetime of the arangod process, so it doesn't need careful optimization. Compare this to the very internals in our code that are called billions of times during the runtime of arangod.

My personal style is to declare variables as close as possible to where they are actually used. This makes things a lot more maintainable IMHO, because I don't have to page up/down several times to see where variables are used. Declare late, exit early.